### DISCUSSION TITLE
Proxmox testing with community

### DISCUSSION THREAD

**Discussion by tormath1** [flatcar/Flatcar#1573] [Show and tell]
A Proxmox image is available here for testing purposes here: http://bincache.flatcar-linux.net/images/amd64/9999.9.100+kai-proxmox-support/flatcar_production_proxmoxve_image.img.bz2.

One can try it following this documentation: https://github.com/flatcar/flatcar-website/pull/337/files or Terraform approach. You can directly suggest your feedback to the documentation or discuss about it in this discussion:

**Comment by ader1990** [comment]
I could succesfully create and run a Flatcar VM with Ignition on Proxmox VE latest (8.2.2):

```bash
export VM_ID=124

qm create $VM_ID --cores 2 --memory 4096 --net0 "virtio,bridge=vmbr0" --ipconfig0 "ip=dhcp"
qm disk import $VM_ID flatcar_production_proxmoxve_image.bin local-lvm

# tell the vm to boot from the imported image
qm set $VM_ID --scsi0 local-lvm:vm-$VM_ID-disk-0
qm set $VM_ID --boot order=scsi0

qm set $VM_ID --ide2 local-lvm:cloudinit

qm set $VM_ID --cicustom "user=local:snippets/userdata"
# userdata is from https://github.com/flatcar/flatcar-website/pull/337/files#diff-47f96e69913e3bf870ccb62bcfad55dda8e04436a2bb2090a1fb7a17f1e57b77R71

qm start $VM_ID
```

Notes: the hostname is not set in this scenario.
Also, if no cloudinit config drive is configured and attached, the OS will drop into an emergency console.

**Reply by tormath1** [reply]
>  Also, if no cloudinit config drive is configured and attached, the OS will drop into an emergency console.

On a regular cloud provider, I would say it's an issue as one can start an instance without user-data and expects afterburn to provision the instance (ssh keys). On Proxmox, there is not such a service so it's mandatory to boot an instance with user-data no?

**Reply by ader1990** [reply]
Similar to Hyper-V, as Proxmox VMs can run without any cloud-config drive, I would expect for Flatcar to boot with autologin enabled.

**Reply by ader1990** [reply]
I suppose it is a matter of choice, mostly security-wise. I just wanted to note what happened when I tried to boot Flatcar without config-drive.

**Reply by tormath1** [reply]
@fhemberger @donch - what do you think regarding this? TL; DR: if no cloudinit config drive is configured and attached, the OS will drop into an emergency console.

I don't have enough Proxmox knowledge to know if that'd be an issue, but maybe we can try to stay consistent with Hyper-V as @ader1990 suggested and allow users to run VMs without cloud-config drive?

**Reply by mcbenjemaa** [reply]
@tormath1 
We should also make it able to use a CDROM ISO labeled as cidata to be accepted.
And not fail cause of no ssh keys or configdrive.

**Reply by donch** [reply]
I would say that without a cloudinit drive, the VM should boot with its default configuration 👍

**Reply by fhemberger** [reply]
Sounds reasonable to fall back to autologin with default configuration. Maybe show a note at login that no ignition config was provided (I don't know from the top of my head what the exact message is right now when you just boot a regular Flatcar ISO).

Users still could run `flatcar-install` at that point. Even if it defeats the point of using a Proxmox image in the first place, this is still better than dropping into an emergency console.

**Reply by tormath1** [reply]
Here's a new image: http://bincache.flatcar-linux.net/images/amd64/9999.0.0+103-kai-proxmox-support/flatcar_production_proxmoxve_image.img (without compression as bzipped was not worth it). It also tests this patch: https://github.com/coreos/afterburn/pull/1128.
Could you folks test if everything is still working as expected + if the instance can boot without a config drive? Thanks :pray:

**Reply by tormath1** [reply]
@ader1990 @donch @fhemberger @mcbenjemaa I gave a try and it boots fine with/without config-drive with the following image :arrow_up:

**Reply by ader1990** [reply]
Hello @tormath1, I verified on my local proxmox and it is working as expected: when cloud-init config drive is not attached, Flatcar boots and there is an autologin. Thanks!

**Reply by donch** [reply]
Hi @tormath1 , I have redone my previous tests and can confirm it still works. Without any cloud-init config drive, the VM starts with autologin enabled. Thank you for your work!

**Reply by mcbenjemaa** [reply]
great job, I can confirm the image boots fine with/without config-drive.

I wish we could support network-config version 2

**Comment by fhemberger** [comment]
@mcbenjemaa

> Can you try to mount `/dev/disk/by-label/cidata` into some tmp folder, and let me know what is inside there?

- `meta-data`: Instance ID
- `network-config`: DNS-Config (YAML)
- `user-data`: The ignition.json config provided
- `vendor-data`: Empty file

**Reply by mcbenjemaa** [reply]
Please, give me an example of the `metadata` and `network-config`

**Reply by fhemberger** [reply]
```
core@flatcar ~/test $ cat meta-data
instance-id: ae1808cd94103dab0a5f9dcbc2b93e0b0f8d9b51
```

```
core@flatcar ~/test $ cat network-config
version: 1
config:
    - type: nameserver
      address:
      - '192.168.178.1'
      search:
      - 'fritz.box'
```

**Reply by mcbenjemaa** [reply]
Thanks

**Comment by mcbenjemaa** [comment]
I used this userdata:
```
	userdata := `#cloud-config
hostname: dummy-vm-hostname
manage_etc_hosts: true
fqdn: dummy.local.com
user: dummy-user
password: $5$6LDowW6p$.RyFu8lVH7Cw3AB.pPS/K2lmB8IczVs99A7gbcUCLV2
ssh_authorized_keys:
  - ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDd1hElre4j44sbmULXyO5j6dRnkRFCMjEGtRSy2SuvFD8WyB5uectcEMvz7ORhQIVbPlz94wFjpSX5wl/gmSKL/7GOyerJo0Y2cvyjJJahuDn+JnIL0tT0HS1pJ5iJqQpxXeOAzMK5Heum+uGw9BzbiUHnRzjJr8Ltx4CAGMfubevD4SX32Q8BTQiaU4ZnGtdHo16pWwRsq1f6/UtL4gDCni9vm8QmmGDRloi/pBn1csjKw+volFyu/kSEmGLWow6NuT6TrhGAbMKas5HfYq0Mn3LGPZL7XjqJQ6CO0TzkG/BNplZT2tiwHtsvXsbePTp4ZUi4dkCMz2xR4eikaI1V dummy@dummy.local
chpasswd:
  expire: False
users:
  - default
package_upgrade: true`
```

But the boot is failing for me.

![Screenshot 2024-10-31 at 13 19 37](https://github.com/user-attachments/assets/b62331f8-f111-479d-868d-74d9e46e2610)

and i'm pretty confused, if I use an ignition in the userdata, it tells the user-data doesn't have `#cloud-config`

**Reply by fhemberger** [reply]
Regular cloud-init config is not supported, only Ignition. Please check the example in the documentation:

https://github.com/enix/flatcar-website/blob/proxmoxve/content/docs/latest/installing/vms/proxmoxve.md#configuring-the-vm-using-ignition

**Reply by mcbenjemaa** [reply]
i got the issue, it's not about user-data but because of the network-config,
it only accepts an version 1
